helm-system-packages-portage.el

```diff
@@ -119,51 +105,37 @@ The caller can pass the list of EXPLICIT packages to avoid re-computing it."
                    (buffer-string)))
    explicit))
 
-(defun helm-system-packages-portage-cache-names ()
-  "Cache all package names."
-  (with-temp-buffer
-    (process-file "eix" nil t nil "--only-names")
-    (buffer-string)))
-
-(defcustom helm-system-packages-portage-column-width 36
-  "Column at which descriptions are aligned, excluding a double-space gap."
+(defun helm-system-packages-portage-cache (display-list)
+  "Cache all package names with descriptions."
+  (let* ((raw (with-temp-buffer
+                (process-file "env" nil '(t nil) nil "EIX_LIMIT=0" "OVERLAYS_LIST=none" "PRINT_COUNT_ALWAYS=never" "eix" "--format" "<category>/<name>\t<description>\n")
+                (sort-lines nil (point-min) (point-max))
+                (buffer-string)))
+         (paired (mapcar (lambda (x) (let ((l (split-string x "\t"))) (cons (car l) (cadr l)))) (split-string raw "\n")))
+         (names (mapcar 'car paired))
+         (descriptions (mapcar (lambda (x) (concat (car x) (make-string (max (- helm-system-packages-column-width (length (car x))) 0) ? ) (cdr x))) paired)))
```

Comment:
Please break all the long lines, thanks!

Reply:
`names` and `descriptions` are supposed to be string buffers for `helm-system-packages--cache-set`.  Is this really working?

Reply:
`helm-init-candidates-in-buffer` accepts either a string or a list of candidates according to its documentation.